src/termio/Exec.zig

```diff
@@ -732,6 +732,19 @@ const Subprocess = struct {
             try env.put("GHOSTTY_RESOURCES_DIR", dir);
         }
 
+        // Unset environment varies set by the snap
+        if (env.get("SNAP") != null) {
+            env.remove("SNAP");
+            env.remove("DRIRC_CONFIGDIR");
+            env.remove("__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS");
+            env.remove("__EGL_VENDOR_LIBRARY_DIRS");
+            env.remove("LD_LIBRARY_PATH");
+            env.remove("LIBGL_DRIVERS_PATH");
+            env.remove("LIBVA_DRIVERS_PATH");
+            env.remove("VK_LAYER_PATH");
+            env.remove("XLOCALEDIR");
```

Comment:
This would unconditionally remove these environment variables from the shell. Reading your launcher file above, you append the various snap paths if the environment variable exists. Wouldn't we just want to remove the snap paths from the environment variables if the user already had them set?

Reply:
No, the problem is the resulting shell in ghostty uses these library paths, drivers, etc for any processes started from ghostty which won't work mixing libs from in the container with host libs. This effectively drops everything needed to start ghostty without leaking those paths into the resulting shell. If a user is setting any of them in /etc or .bashrc, etc those will be honored by the shell anyway.